src/matrix.jl

```diff
@@ -41,6 +41,52 @@ getindex(M::IntervalMatrix, i::Int) = getindex(M.mat, i)
 setindex!(M::IntervalMatrix, X, inds...) = setindex!(M.mat, X, inds...)
 copy(M::IntervalMatrix) = IntervalMatrix(copy(M.mat))
 
+# undef initializer, eg. IntervalMatrix{Float64}(undef, 2, 2)
+function IntervalMatrix{T}(u::UndefInitializer, m, n) where {T}
+    mat = Matrix{Interval{T}}(undef, m, n)
+    return IntervalMatrix(mat)
+end
+
+"""
+    IntervalMatrix(C::MT, S::MT) where {T, MT<:AbstractMatrix{T}}
+
+Return an interval matrix such that the center and radius of the intervals
+is given by the matrices `C` and `S` respectively.
+
+### Input
+
+- `C` -- center matrix
+- `S` -- radii matrix
+
+### Output
+
+An interval matrix `M` such that `M[i, j]` corresponds to the interval whose center
+is `C[i, j]` and whose radius is `S[i, j]`, for each `i` and `j`. That is,
+``M = C + [-S, S]``.
+
+### Notes
+
+The radii matrix, should be nonnegative, i.e. `S[i, j] ≥ 0` for each `i` and `j`.
+This assumption is not checked in this constructor.
+"""
+function IntervalMatrix(C::MT, S::MT) where {T, MT<:AbstractMatrix{T}}
+    size(C) == size(S) || throw(ArgumentError("the sizes of the center matrix and the " *
+                                "radii matrix should match, but they are $(size(C)) " *
+                                "and $(size(S)) respectively"))
+    m, n = size(C)
+    M = IntervalMatrix{T}(undef, m, n)
+
+    @inbounds for j in 1:n
+        for i in 1:m
+            inf_ij = C[i, j] - S[i, j]
+            sup_ij = C[i, j] + S[i, j]
+            M[i, j] = Interval{T}(inf_ij, sup_ij)
+        end
+    end
```

Comment:
I think this is not the correct semantics because you do the operations in `T` (typically floating-point numbers) and then pass the result to `IntervalArithmetic`. We should rather let `IntervalArithmetic` do the computations including the rounding.

Reply:
Ok, i agree. There is a problem with `T`. When i wrote this, i was having in mind the behavior of `IntervalArithmetic.Interval` when you pass an integer: it automatically converts to float, 

```julia
julia> Interval(1)
[1, 1]

julia> eltype(ans)
Float64
```

On the other hand, `IntervalMatrix{T}(undef, ...)` is problematic, if `T` is an integer. Then i propose to use 

```julia
    F = eltype(Interval(one(T)))
    M = IntervalMatrix{F}(undef, m, n)
```
and the behavior is now the same as in the scalar interval case:

```julia
julia> IntervalMatrix([1 2; 3 4], [1 2; 4 5])
2×2 IntervalMatrix{Float64,Interval{Float64},Array{Interval{Float64},2}}:
  [0, 2]   [0, 4]
 [-1, 7]  [-1, 9]
```

Reply:
You want to use this `Interval` constructor:
```julia
M[i, j] = C[i, j] ± S[i, j]
```

Reply:
I think that's the same as lines 85-87 in the PR.

Reply:
..  well not quite, because `C[i, j] ± S[i, j]` checks that the interval is well formed since it uses `interval` (see [here](https://github.com/JuliaIntervals/IntervalArithmetic.jl/blob/master/src/intervals/intervals.jl#L174)), while `Interval` does not perform such checks (see  https://juliaintervals.github.io/IntervalArithmetic.jl/latest/construction/ for the different constructors).

Reply:
So i agree to change those lines to using `C[i, j] ± S[i, j]` and remove the comment on the assumption (https://github.com/JuliaReach/IntervalMatrices.jl/pull/107/files#diff-8966c035a38e8674351558a682403bcdR64)

Reply:
If you scroll down in your link, you will find that ± performs rounding:

```
m ± r

The ± operator (typed as \pm<TAB>) creates the interval with midpoint m
and radius r, and is equivalent to (m - r) .. (m + r):

julia> 1 ± 0.1
Interval(0.8999999999999999, 1.1)
```

Reply:
```julia
julia> Interval(1.0-0.1, 1.0+0.1)
[0.9, 1.10001]

julia> Interval(1.0±0.1)
[0.899999, 1.10001]
```

Reply:
So which constructor do you suggest to use?

Reply:
Also, what is the purpose of wrapping with `Interval` after using `±` as in your last comment? (`Interval(1.0±0.1)`)

Reply:
`C[i, j] ± S[i, j]`, as you said :+1:

Reply:
> Also, what is the purpose of wrapping with Interval

True, that is redundant.